### PR TITLE
Fixes to allow OSX build to compile and run without crashing

### PR DESCRIPTION
**Pull request by tmiw**
This is the second part of the needed changes to get OSX builds working again (the first part involves Codec2 changes; that pull request is at https://github.com/drowe67/codec2/pull/66). To do this, build_osx.sh was created as well as various CMake files fixed to ensure that consistent OSX compiler/linker flags are used throughout all prerequisites.

### CODE REVIEW

**Comment by drowe67** [comment]
Great work @tmiw - I've asked Mark to test on his machine.  It would also be nice to configure Travis to build this to ensure we don't break the OSX build as freedv-gui evolves.

File: build_osx.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+cd hamlib-code && git checkout master && git pull
+./bootstrap
+CFLAGS="-g -O2 -mmacosx-version-min=10.9" CXXFLAGS="-g -O2 -mmacosx-version-min=10.9" ./configure --disable-shared --prefix $HAMLIBDIR
+make
+make install
+
+# First build and install vanilla codec2 as we need -lcodec2 to build LPCNet
+cd $FREEDVGUIDIR
+git clone https://github.com/drowe67/codec2.git
+cd codec2 && git checkout master && git pull
```
Comment (drowe67) [inline build_osx.sh]: @tmiw do we need to clone from tmiw:master here to test this PR?

Comment (tmiw) [reply build_osx.sh]: Correct.

**Comment by darksidelemm** [comment]
I can confirm that this PR lets me build under OSX!
Note my comments on the related codec2 PR here https://github.com/drowe67/codec2/pull/66

I haven't done any testing of the resultant build, but it does open, so I'd say that's good enough to approve this PR.

File: build_osx.sh (hunk 1 above)
Comment (db4ple) [reply build_osx.sh]: I did this (by changing the script in my own local git repo), and it worked.

**Comment by drowe67** [comment]
Oops I think I posted a comment to the wrong PR.   Here's my test request: https://github.com/drowe67/codec2/tree/master/wav for this build of freedv-gui

**Pull request merged**